oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/NodeDocument.java

```diff
@@ -610,6 +610,9 @@ int purgeUncommittedRevisions(final int clusterId, final int batchSize,
             if (!Utils.isCommitted(commitValue) && r.getClusterId() == clusterId && olderThanLastWrittenRootRevPredicate.test(r)) {
                 uniqueRevisions.add(r);
                 removeBranchCommit(op, r);
+                if (op.getId().equals(Utils.getIdFromPath("/"))) {
+                    removeCommitRoot(op, r);
+                }
```

Comment:
```suggestion
                if (op.getId().equals(Utils.getIdFromPath(Path.ROOT))
                        && getLocalCommitRoot().containsKey(r)) {
                    removeCommitRoot(op, r);
                }
```

While this is somewhat of a detail, I think it would make it less intrusive if `removeCommitRoot` was only called when the revision `r` is actually on that document (`this`). Otherwise this is added to many unnecessary cases (where it wouldn't necessarily do harm, but be unnecessary). Wdyt?

Reply:
Done. It is fine for me.